src/Microsoft.Health.SqlServer/Features/Client/SqlConnectionWrapper.cs

```diff
@@ -17,55 +16,74 @@ public class SqlConnectionWrapper : IDisposable
         private readonly bool _enlistInTransactionIfPresent;
         private readonly SqlTransactionHandler _sqlTransactionHandler;
         private readonly SqlCommandWrapperFactory _sqlCommandWrapperFactory;
+        private readonly ISqlConnectionFactory _sqlConnectionFactory;
+
+        private SqlConnection _sqlConnection;
+        private SqlTransaction _sqlTransaction;
 
         public SqlConnectionWrapper(
-            SqlServerDataStoreConfiguration configuration,
             SqlTransactionHandler sqlTransactionHandler,
             SqlCommandWrapperFactory sqlCommandWrapperFactory,
+            ISqlConnectionFactory sqlConnectionFactory,
             bool enlistInTransactionIfPresent)
         {
-            EnsureArg.IsNotNull(configuration, nameof(configuration));
             EnsureArg.IsNotNull(sqlTransactionHandler, nameof(sqlTransactionHandler));
             EnsureArg.IsNotNull(sqlCommandWrapperFactory, nameof(sqlCommandWrapperFactory));
+            EnsureArg.IsNotNull(sqlConnectionFactory, nameof(sqlConnectionFactory));
 
             _sqlTransactionHandler = sqlTransactionHandler;
             _enlistInTransactionIfPresent = enlistInTransactionIfPresent;
             _sqlCommandWrapperFactory = sqlCommandWrapperFactory;
+            _sqlConnectionFactory = sqlConnectionFactory;
+
+            InitializeAsync();
+        }
+
+        public SqlConnection SqlConnection
+        {
+            get { return _sqlConnection; }
+        }
+
+        public SqlTransaction SqlTransaction
+        {
+            get
+            {
+                return _sqlTransaction;
+            }
+        }
```

Comment:
should not do async void

Reply:
It's not really returning anything. What should the return type be?

Reply:
never mind, just the Task, right?